tests/test_font.py

```diff
@@ -11,402 +11,265 @@
-            font("italic normal 12pt Comic Sans"),
-            "Comic Sans",
-            12,
-            ITALIC,
-            NORMAL,
-            NORMAL,
-        )
-
-        self.assertFont(
-            font("italic small-caps normal 12pt Comic Sans"),
-            "Comic Sans",
-            12,
-            ITALIC,
-            SMALL_CAPS,
-            NORMAL,
-        )
-
-    def test_style(self):
-        self.assertFont(
-            font("italic 12pt Comic Sans"), "Comic Sans", 12, ITALIC, NORMAL, NORMAL
-        )
-
-        self.assertFont(
-            font("oblique 12pt Comic Sans"), "Comic Sans", 12, OBLIQUE, NORMAL, NORMAL
-        )
-
-        with self.assertRaises(ValueError):
-            font("wiggly small-caps bold 12pt Comic Sans")
-
-    def test_variant(self):
-        self.assertFont(
-            font("italic small-caps 12pt Comic Sans"),
-            "Comic Sans",
-            12,
-            ITALIC,
-            SMALL_CAPS,
-            NORMAL,
-        )
-
-        with self.assertRaises(ValueError):
-            font("italic wiggly bold 12pt Comic Sans")
-
-    def test_weight(self):
-        self.assertFont(
-            font("italic small-caps bold 12pt Comic Sans"),
-            "Comic Sans",
-            12,
-            ITALIC,
-            SMALL_CAPS,
-            BOLD,
-        )
 
-        with self.assertRaises(ValueError):
-            font("italic small-caps wiggly 12pt Comic Sans")
+@pytest.mark.parametrize("kwargs", [{}, {"variant": SMALL_CAPS}])
+def test_make_normal_variant(kwargs):
+    f = Font("Comic Sans", 12, **kwargs)
+    assert_font(f.normal_variant(), "Comic Sans", 12, NORMAL, NORMAL, NORMAL)
 
-    def test_duplicates(self):
-        with self.assertRaises(ValueError):
-            font("oblique italic 12pt Comic Sans")
 
-        with self.assertRaises(ValueError):
-            font("italic small-caps oblique 12pt Comic Sans")
+def test_make_small_caps():
+    f = Font("Comic Sans", 12)
+    assert_font(f.small_caps(), "Comic Sans", 12, NORMAL, SMALL_CAPS, NORMAL)
 
-        with self.assertRaises(ValueError):
-            font("italic small-caps bold small-caps 12pt Comic Sans")
 
-        with self.assertRaises(ValueError):
-            font("bold bold 12pt Comic Sans")
+@pytest.mark.parametrize(
+    "weight, result", [(BOLD, BOLD), ("bold", BOLD), ("something else", NORMAL)]
+)
+def test_weight(weight, result):
+    assert_font(
+        Font("Comic Sans", 12, weight=weight),
+        "Comic Sans",
+        12,
+        NORMAL,
+        NORMAL,
+        result,
+    )
+
+
+@pytest.mark.parametrize("kwargs", [{}, {"weight": BOLD}])
+def test_make_normal_weight(kwargs):
+    f = Font("Comic Sans", 12, **kwargs)
+    assert_font(f.normal_weight(), "Comic Sans", 12, NORMAL, NORMAL, NORMAL)
+
+
+def test_make_bold():
+    f = Font("Comic Sans", 12)
+    assert_font(f.bold(), "Comic Sans", 12, NORMAL, NORMAL, BOLD)
+
+
+###############
+# Parsing fonts
+###############
```

Comment:
The fact that there's two "types" of tests in this file suggests it's probably worth breaking the test file into two parts. The fact that you need to put a "parse" prefix on all the test names is the giveaway for this.

Reply:
Sure, can do.